frontend/src/components/nodes/NodeRow.tsx

```diff
@@ -9,7 +9,46 @@ interface Props {
   node: N.NodeInfo;
 }
 
-export default class extends React.PureComponent<Props> {
+interface State {
+  statusExplanation?: string;
+}
+
+export default class extends React.PureComponent<Props, State> {
+  state: State = {};
+
+  NodeStatusMessage = (status: string) => {
+    let explanation;
+    switch (status) {
+      case "AwaitingPeers":
+        explanation = "Waiting for peers";
+        break;
+      case "HeaderSync":
+        explanation = "Syncing headers";
+        break;
+      case "BlockSync":
+        explanation = "Syncing blocks";
+        break;
+      case "StateSync":
+        explanation = "Syncing state";
+        break;
+      case "StateSyncDone":
+        explanation = "State sync is done";
+        break;
+      case "BodySync":
+        explanation = "Syncing body";
+        break;
+      case "NoSync":
+      default:
+        explanation = undefined;
+    }
+    return explanation;
+  };
```

Comment:
Hashmap would be much shorter, and no need in saving it to state, just fetch the description by the status identifier right on render.